lib/dialects/abstract/query-generator.js

```diff
@@ -198,7 +203,39 @@ module.exports = (function() {
           valueQuery += ' RETURNING *';
           emptyQuery += ' RETURNING *';
         } else if (!!this._dialect.supports.returnValues.output) {
-          outputFragment = ' OUTPUT INSERTED.*';
+          //Loop through each field type for this table and create a variable table for the output of INSERTED table
+          if (modelAttributes) {
+            tmpTableForMSSQL = 'declare @tmp table (<%= columns %>); '
+
+            for (key in modelAttributes){
+              attribute = modelAttributes[key];
+              if (attribute.type.toString().toUpperCase() != 'VIRTUAL'){
+                if (tmpColumns.length > 0){
+                  tmpColumns += ','
+                  outputColumns += ','
+                }
+
+                //This is a hack, we should determine the size of the field and use that.
+                //However, having a max doesn't hurt,
+                if(attribute.type.toString().toUpperCase().indexOf('NVARCHAR') > -1)
+                {
+                  tmpColumns += attribute.fieldName + ' NVARCHAR(MAX)'
```

Comment:
Shouldn't this use `attribute.field`, rather than `attribute.fieldName`? The docs  briefly mention this in [model definition](http://sequelize.readthedocs.org/en/latest/docs/models-definition/) - see the last couple of lines in the second code sample. I'm having issues where my field alias is being used as the column name in the temporary table instead of the actual column name, resulting in the following error:

```
Unhandled rejection SequelizeDatabaseError: Invalid column name 'id'.
    at module.exports.Query.formatError (\\path\to\project\node_modules\sequelize\lib\dialects\mssql\query.js:211:1
2)
```

Where my model definition looks like:

```javascript
var Customer = orm.define('customers', {
	id: {
		type: Sequelize.INTEGER,
		primaryKey: true,
		autoIncrement: true,
		field: 'CustomerID'
	}
	// [...]
}
```

Same for lines 225, 227, 370, 373, 375